### DISCUSSION TITLE
read_power_activities: Annotated 0 pin activities.

### DISCUSSION THREAD

**Discussion by yyan7223** [The-OpenROAD-Project/OpenROAD-flow-scripts#2494] [Q&A] [answered]
I'm currently trying to re-estimate the power of my design with a VCD file (RTL level), using API `read_power_activities`, but it reports `Annotated 0 pin activities` no matter what `-scope` I pass to it:
<img width="740" alt="1729585282668" src="https://github.com/user-attachments/assets/0d7e308b-1d0d-4c7e-aa64-204087772c92">
I find that most of signals' names in `6_final.v` and `VCD` are different and that is probably why no pin activities are annotated. Is that because my `VCD` file is RTL level but `6_final.v` is gate-level? If so, how should I do the gate-level simulation and generate `VCD` file? I'm new to OpenRoad, any appreciate to any helps.

Best regards,
Yufei

**Comment by maliberty** [comment, marked as answer]
You should run your RTL simulation on the 6_final.v and generate the VCD from that.

**Reply by yyan7223** [reply]
Yeah it works, thank you!
Another question, is there any API to request the power distribution of each grid just like shown in the heatmap?
![image](https://github.com/user-attachments/assets/924ad805-a28f-422f-844a-4b1e710fa2d8)

Best regards,
Yufei

**Reply by maliberty** [reply]
See https://openroad.readthedocs.io/en/latest/main/src/gui/README.html#dump-heatmap-to-file
